add binding for mqtt

mhusinsky opened the pull request:
this is initial work for binding MQTT to public channels. 

it's mainly based on the OSC bindings and works quite nicely -  but for now is limited to publishing strings only.
i would need your input as i'm not entirely sure what's the best way to set it up in a generic way for other types (at least bytes for "raw" sending of data).

mhusinsky committed:
initial commit for mqtt binding

mhusinsky committed:
renamed category to MQTT2 to align with naming in other packs (OSC, MIDI).

cleanup in helppatch

mhusinsky committed:
made publishing/receiving to channels now adaptive/generic (only string implementation provided for now).

note: the binding is buggy for VVVV versions 7.1-42 and later - needs fixing

mhusinsky committed:
fix for "made publishing/receiving to channels now adaptive/generic (only string implementation provided for now)."

mhusinsky committed:
Merge branch 'mqttbinding' of https://github.com/mhusinsky/VL.IO.MQTT into mqttbinding

azeno commented:
By just looking at your screenshot I wonder if the type you feed to GetTypeByName is correct? Its name suggests that it's non generic / has no type parameters while further down (the node that crashes) surely expects a generic type - it's the whole point to replace some type parameter T with something concrete like String.

mhusinsky committed:
added null checks and connection checks for sending MQTT messages.

this became necessary as the public channels apparently try to send an initial message immediately after the patch started, before the connection was established.

publish nodes now don't throw an exception when broker is not connected

gregsn committed:
help patch showing a round trip. changing the value in one public channel will change the other.
